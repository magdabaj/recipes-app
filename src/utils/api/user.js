const api = 'https://afternoon-plateau-23579.herokuapp.com/user';
// const api = 'http://localhost:5000/user'

export const signUpUser = async (email, password) => {
    return fetch(`${api}/signup`, {
        method: 'POST',
        body: JSON.stringify({ email, password }),
        headers: { 'Content-Type': 'application/json' },
        credentials: 'include',
    }).then(res => res.json());
};

export const signInUser = async (email, password) => {
    return fetch(`${api}/signin`, {
        method: 'POST',
        body: JSON.stringify({ email, password }),
        headers: { 'Content-Type': 'application/json' },
        credentials: 'include',
    }).then(res => res.json());
};

export const signOutUser = async () => {
    return fetch(`${api}/signout`, {
        headers: { 'Content-Type': 'application/json' },
        credentials: 'include',
    }).then(res => res.json());
};

export const authenticateUser = async () => {
    return fetch(`${api}/authenticated`, {
        headers: { 'Content-Type': 'application/json' },
        credentials: 'include',
    }).then(res => res.json());
};
