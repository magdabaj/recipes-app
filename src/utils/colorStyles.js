const colorStyles = {
    darkRed: '#5D001E',
    lightGrey: '#E3E2DF',
    lightPink: '#E3AFBC',
    mediumPink: '#EE4C7C',
    darkPink: '#9A1750',
};

export default colorStyles;
