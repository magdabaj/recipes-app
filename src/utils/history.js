// import { createBrowserHistory } from 'history';
//
// const history = createBrowserHistory();

import { createMemoryHistory } from "history";

const history = createMemoryHistory()

export default history;
