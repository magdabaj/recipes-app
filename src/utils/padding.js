const basePadding = 8;

const padding = {
    extraSmall: `${basePadding / 4}px`, // 2px
    small: `${basePadding / 2}px`, // 4px
    base: `${basePadding}px`, // 8px
    medium: `${basePadding * 1.5}px`, // 12px
    large: `${basePadding * 2}px`, // 16px
}

export default padding;