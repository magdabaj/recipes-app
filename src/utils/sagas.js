import { all } from 'redux-saga/effects';
import { watchRecipesByTagLoad, watchRecipesLoad, watchTagsLoad } from "../containers/HomePage/saga";

export default function* rootSaga() {
    yield all([
        watchRecipesLoad(),
        watchTagsLoad(),
        watchRecipesByTagLoad()
    ]);
}