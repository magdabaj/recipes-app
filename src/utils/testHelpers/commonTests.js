const commonTests = (render, useFirstChildAsSnapshot = true) => {
    it('does not log errors in console', () => {
        const spy = jest.spyOn(global.console, 'error')
        render()
        expect(spy).not.toHaveBeenCalled()
    })

    it('should render and match the snapshot',  () => {
        const { container } = render()
        expect(container).not.toBeNull()
        if(useFirstChildAsSnapshot) expect(container.firstChild).not.toBeNull()
        expect(
            useFirstChildAsSnapshot ? container.firstChild : container,
        ).toMatchSnapshot()
    });
}

export default commonTests;