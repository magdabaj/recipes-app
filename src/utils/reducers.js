import { combineReducers } from 'redux';
import { connectRouter } from "connected-react-router";
import homePageReducer from "../containers/HomePage/reducer";
import loginContainerReducer from "../containers/LoginContainer/reducer";
import userRecipesContainerReducer from "../containers/UserRecipesContainer/reducer";
import recipesFormContainerReducer from "../containers/RecipesFormContainer/reducer";
import recipeContainerReducer from "../containers/RecipeContainer/reducer";
import appReducer from '../containers/App/reducer';
import history from "./history";

const rootReducer = combineReducers({
    homePageReducer,
    loginContainerReducer,
    userRecipesContainerReducer,
    recipesFormContainerReducer,
    recipeContainerReducer,
    appReducer,
    router: connectRouter(history),
})

export default rootReducer;