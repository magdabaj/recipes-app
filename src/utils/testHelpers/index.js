import { Router } from 'react-router-dom'
import { createMemoryHistory } from "history";
import store from "../store";
import {Provider} from "react-redux";
import React from "react";
import { render } from "@testing-library/react";
import {createStore} from "redux";
// import {reducer} from "react-toastify/dist/hooks/toastContainerReducer";

// const store  = createStore(rootReducer, initialState={})
const memoryHistory = createMemoryHistory({
    initialEntries: ['/'],
})

export const withStoreAndRouter = component =>
    <Router history={memoryHistory}>
        <Provider store={store}>
            {component}
        </Provider>
    </Router>

export const renderWithRouter = (
    ui,
    {
        history = memoryHistory,
    } = {}
) => (
    {
        ...render(<Router history={history}>{ui}</Router>),
        history,
    }
)

export const renderWithRouterKentCDodds = (
    ui,
        {
            route='/',
            history = createMemoryHistory({initialEntries: [route]}),
            ...renderOptions
        } = {}
) => {
    // eslint-disable-next-line react/prop-types
    function Wrapper({children}) {
        return <Router history={history}>
            {children}
        </Router>
    }

    return {
        ...render(ui, {wrapper: Wrapper, ...renderOptions})
    }
}

export const renderWithProviderKentCDodds = (
        ui,
        reducer,
        {initialState, store = createStore(reducer, initialState), ...options} = {},
    ) => {
    // eslint-disable-next-line react/prop-types
    function Wrapper({children}) {
        return <Provider store={store}>
            {children}
        </Provider>
    }
    return render(ui, {wrapper: Wrapper, ...options})
}