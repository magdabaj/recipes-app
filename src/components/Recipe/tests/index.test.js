import Recipes from "../index";
import {fakeUser} from "../../../utils/testHelpers/fixtures/user";
import {fakeRecipes} from "../../../utils/testHelpers/fixtures/recipes";
import commonTests from "../../../utils/testHelpers/commonTests";
import React from "react";
import {renderWithRouter} from "../../../utils/testHelpers";

const sendRatingMocked = jest.fn()

const renderRecipe = () =>
    renderWithRouter(<Recipes user={fakeUser} commentsNumber={5} sendRating={sendRatingMocked} recipes={fakeRecipes}/>)

commonTests(renderRecipe)

