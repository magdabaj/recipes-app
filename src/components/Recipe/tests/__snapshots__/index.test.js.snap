// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render and match the snapshot 1`] = `
<div
  class="RecipeContainer-pn5chn-0 iMuICT"
>
  <div
    class="Image-sc-1abb6jd-0 bmqIel"
  >
    <div
      class="RecipeTitle-hqztkb-0 EWXAm"
    >
      <a
        href="test"
        rel="noreferrer"
        target="_blank"
      >
        test
      </a>
    </div>
    <div
      class="RecipeRatingContainer-sc-5jz1kw-0 lozgei"
    >
      <div
        class="AddRatingButton-zt5srs-0 htnUHB"
      >
        <span
          class="Icon-sc-1rpttgp-0 gzJki"
        >
          <svg
            class="heart-icon"
            fill="currentColor"
            height="16px"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="16px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
            />
          </svg>
          <svg
            class="heart-icon"
            fill="currentColor"
            height="16px"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="16px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
            />
          </svg>
          <svg
            class="heart-icon"
            fill="currentColor"
            height="16px"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="16px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
            />
          </svg>
          <svg
            class="heart-icon"
            fill="currentColor"
            height="16px"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="16px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
            />
          </svg>
          <svg
            class="heart-icon"
            fill="currentColor"
            height="16px"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="16px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
            />
          </svg>
        </span>
      </div>
      <div
        class="AddRatingButton-zt5srs-0 htnUHB"
      >
        Komentarze (
        5
        )
        <span
          class="Icon-sc-1rpttgp-0 gzJki"
        >
          <svg
            class="heart-icon"
            fill="currentColor"
            height="16px"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 576 512"
            width="16px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M486.41 107.57c-76.93-50.83-179.18-62.4-264.12-47.07C127.26-31.16 20.77 11 0 23.12c0 0 73.08 62.1 61.21 116.49-86.52 88.2-45.39 186.4 0 232.77C73.08 426.77 0 488.87 0 488.87c20.57 12.16 126.77 54.19 222.29-37 84.75 15.23 187 3.76 264.12-47.16 119.26-76.14 119.65-220.61 0-297.15zM294.18 404.22a339.53 339.53 0 0 1-88.11-11.37l-19.77 19.09a179.74 179.74 0 0 1-36.59 27.39A143.14 143.14 0 0 1 98 454.06c1-1.78 1.88-3.56 2.77-5.24q29.67-55 16-98.69c-32.53-25.61-52-58.34-52-94.13 0-82 102.74-148.43 229.41-148.43S523.59 174 523.59 256 420.85 404.22 294.18 404.22zM184.12 291.3a34.32 34.32 0 0 1-34.8-33.72c-.7-45.39 67.83-46.38 68.52-1.09v.51a34 34 0 0 1-33.72 34.32zm73.77-33.72c-.79-45.39 67.74-46.48 68.53-1.19v.61c.39 45.08-67.74 45.57-68.53.58zm143.38 33.72a34.33 34.33 0 0 1-34.81-33.72c-.69-45.39 67.84-46.38 68.53-1.09v.51a33.89 33.89 0 0 1-33.72 34.32z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div>
      <a
        href="test"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="test"
          class="Recipe__Img-sc-1bhnvxm-0 jJscgp"
          src="test"
        />
      </a>
    </div>
    <div
      class="source-website"
    >
       Pobrano z: 
      test
    </div>
    <div
      class="source-website"
    >
      <a
        class="source-website--link"
        href="test"
        rel="noreferrer"
        target="_blank"
      >
        Kliknij zeby zobaczyc przepis
      </a>
    </div>
  </div>
</div>
`;
